src/uu/paste/src/paste.rs

```diff
@@ -118,39 +166,74 @@ fn paste(
         ));
     }
 
-    let delimiters: Vec<char> = unescape(delimiters).chars().collect();
-    let mut delim_count = 0;
-    let mut delim_length = 1;
-    let stdout = stdout();
-    let mut stdout = stdout.lock();
+    // Precompute instead of doing this inside the loops
+    let mut delimiters_encoded_option = {
+        let delimiters_unescaped = unescape(delimiters).chars().collect::<Vec<_>>();
+
+        let number_of_delimiters = delimiters_unescaped.len();
+
+        if number_of_delimiters > 0_usize {
+            let mut vec = Vec::<Box<[u8]>>::with_capacity(number_of_delimiters);
+
+            {
+                // a buffer of length four is large enough to encode any char
+                let mut buffer = [0_u8; 4_usize];
+
+                for ch in delimiters_unescaped {
+                    let delimiter_encoded = ch.encode_utf8(&mut buffer);
+
+                    vec.push(Box::from(delimiter_encoded.as_bytes()));
+                }
+            }
```

Comment:
Why do you put this code inside a block?

Reply:
Just do denote that limit the scope of `buffer`. I'll remove the block.